### PR TITLE
SetOperatorStatusCondition update transition time for existing

### PR DESCRIPTION
**Pull request by michaelgugino**
This commit ensures whenever there are differences between
existing and new status, we update the transition time with
current time.

### CODE REVIEW

File: lib/resourcemerge/os.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func SetOperatorStatusCondition(conditions *[]configv1.ClusterOperatorStatusCond
 
 	if existingCondition.Status != newCondition.Status {
 		existingCondition.Status = newCondition.Status
-		existingCondition.LastTransitionTime = newCondition.LastTransitionTime
+		existingCondition.LastTransitionTime = metav1.NewTime(time.Now())
```
Comment (runcom) [inline lib/resourcemerge/os.go]: `metav1.Now()`?

Comment (michaelgugino) [reply lib/resourcemerge/os.go]: https://github.com/openshift/cluster-version-operator/pull/153/files#diff-7b07a5cf0629a7b019ac699efb7cfe94R47

This is the same method used to generate time elsewhere in this file.

Comment (runcom) [reply lib/resourcemerge/os.go]: yeah, just pointing out that there's a direct one, just a nit

**Comment by abhinavdahiya** [comment]
Do you mind explaining why using the last transition time from new condition is not correct. The user must have set the time correctly already right?

**Comment by michaelgugino** [comment]
> Do you mind explaining why using the last transition time from new condition is not correct. The user must have set the time correctly already right?

@abhinavdahiya 
No, we're not setting the last update time.  I suspect nobody else is either.  I don't see much use for this method if it's not going to set the transition time on transitions.

**Pull request merged**